ArduCopter/mode_autorotate.cpp

```diff
@@ -134,25 +199,29 @@ void ModeAutorotate::run()
 
             }
 
-            // Slowly change the target head speed until the target head speed matches the parameter defined value
-            if (g2.arot.get_rpm() > HEAD_SPEED_TARGET_RATIO*1.005f  ||  g2.arot.get_rpm() < HEAD_SPEED_TARGET_RATIO*0.995f) {
+            if(!hover_autorotation){ 
+			// Slowly change the target head speed until the target head speed matches the parameter defined value
+           if (g2.arot.get_rpm() > HEAD_SPEED_TARGET_RATIO*1.005f  ||  g2.arot.get_rpm() < HEAD_SPEED_TARGET_RATIO*0.995f) {
                 _target_head_speed -= _hs_decay*G_Dt;
             } else {
                 _target_head_speed = HEAD_SPEED_TARGET_RATIO;
             }
-
-            // Set target head speed in head speed controller
-            g2.arot.set_target_head_speed(_target_head_speed);
-
-            // Run airspeed/attitude controller
-            g2.arot.set_dt(G_Dt);
-            g2.arot.update_forward_speed_controller();
-
-            // Retrieve pitch target
-            _pitch_target = g2.arot.get_pitch();
-
-            // Update controllers
-            _flags.bad_rpm = g2.arot.update_hs_glide_controller(G_Dt); //run head speed/ collective controller
+			  g2.arot.get_collective_minimum_drag(motors->get_coll_mid());
+               // Set target head speed in head speed controller
+               g2.arot.set_target_head_speed(_target_head_speed);
+               // Run airspeed/attitude controller
+               g2.arot.set_dt(G_Dt);
```

Comment:
@Gone4Dirt why is the set_dt called in each of the cases for the state machine which was in the original code.  Why not call it when autorotate is initialized?  Also why is the update_hs_glide_controller have it as an argument.  again why not set it when initialized?

Reply:
Yes, it can be done as you say.  The intent was to continually set an updated, measured value. But I never got to that.  So as it is passing a constant it be changed as you suggest.

Reply:
That is a good point. We probably should consider doing that in light of the more resources being required with is slowing down the loop time.

Reply:
This has been fixed. Moved set dt to the start of the run method.